src/Header/Header.js

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from "react";
+import "./Header.css";
+import logoImg from "../images/logo.svg";
+
+function Header() {
+  let [userInfo, setUserInfo] = useState();
+  const invisible = { display: "none" };
+  const visible = { display: "inherit" };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await fetch("https://bootcamp-api.codeit.kr/api/sample/user");
+      const data = await response.json();
+      setUserInfo(data);
+    };
+    fetchData();
+  }, []);
+
+  if (!userInfo) {
+    return null;
+  }
+  console.log(userInfo.id);
+  return (
+    <header>
+      <div>
+        <a href="/">
+          <img src={logoImg} alt="로고" />
+        </a>
+        <a href="./signin/signin.html" style={userInfo.id !== 1 ? visible : invisible}>
+          로그인
+        </a>
+        <div className="user-info" style={userInfo.id === 1 ? visible : invisible}>
+          <img src={userInfo["profileImageSource"]} alt="prifile" />
+          <h6>{userInfo["email"]}</h6>
+        </div>
```

Comment:
조건부렌더링을 하기 위해, style 속성을 이용하셨군요. 

리액트에서 조건부 렌더링은 JSX 자체에 삼항조건연산자를 적용하여 하시는 것이 일반적이고 더 간단합니다.
```suggestion
    {userInfo ? (
      <div className='user-info'>
        <img
          src={userInfo["profileImageSource"]}
          alt='profile'
        />
        <h6>{userInfo["email"]}</h6>
      </div>
    ) : (
      <a href='./signin/signin.html'>로그인</a>
    )}
```

react 공식문서의 조건부 렌더링 페이지를 읽어보실 것은 추천드립니다.
https://ko.react.dev/learn/conditional-rendering

Reply:
그리고 
객체의 프로퍼티 접근 방법으로서
괄호표기법인 `userInfo["profileImageSource"]` 대신에, 
점 표기법인 `userInfo.profileImageSource` 를 사용해보세요. 

점 표기법이 더 간편해서 일반적으로 쓰입니다.

다만, 키 값이 변수이거나 숫자로 시작하거나 공백이 있는 경우는 괄호표기법만 가능하므로 그런 경우에 한하여 괄호표기법을 사용하실 수 있습니다.

객체 프로퍼티 접근 방법에 대해 자세한 내용은 아래 링크 참고해보세요. 
https://ko.javascript.info/object#ref-271